docs/design/admission_control_resource_quota.md

```diff
@@ -54,109 +58,139 @@ const (
+| pods | Total number of active pods where phase is pending or active.  |
 | services | Total number of services |
 | replicationcontrollers | Total number of replication controllers |
 | resourcequotas | Total number of resource quotas |
+| secrets | Total number of secrets |
+| persistentvolumeclaims | Total number of persistent volume claims |
 
-Any resource that is not part of core Kubernetes must follow the resource naming convention prescribed by Kubernetes.
+If a third-party wants to track additional resources, it must follow the resource naming conventions prescribed
+by Kubernetes.  This means the resource must have a fully-qualified name (i.e. mycompany.org/shinynewresource)
 
-This means the resource must have a fully-qualified name (i.e. mycompany.org/shinynewresource)
+## Resource Requirements: Requests vs Limits
 
-If the incoming request does not cause the total usage to exceed any of the enumerated hard resource limits, the plug-in will post a
-**ResourceQuotaUsage** document to the server to atomically update the observed usage based on the previously read
-**ResourceQuota.ResourceVersion**.  This keeps incremental usage atomically consistent, but does introduce a bottleneck (intentionally)
-into the system.
+If a resource supports the ability to distinguish between a request and a limit for a resource,
+the quota tracking system will only cost the request value against the quota usage.  If a resource
+is tracked by quota, and no request value is provided, the associated entity is rejected as part of admission.
 
-To optimize system performance, it is encouraged that all resource quotas are tracked on the same **ResourceQuota** document.  As a result,
-its encouraged to actually impose a cap on the total number of individual quotas that are tracked in the **Namespace** to 1 by explicitly
-capping it in **ResourceQuota** document.
+For an example, consider the following scenarios relative to tracking quota on CPU:
 
-## kube-apiserver
+| Pod | Container | Request CPU | Limit CPU | Result |
+| --- | --------- | ----------- | --------- | ------ |
+| X | C1 | 100m | 500m | The quota usage is incremented 100m |
+| Y | C2 | 100m | none | The quota usage is incremented 100m |
+| Y | C2 | none | 500m | The quota usage is incremented 500m since request will default to limit |
+| Z | C3 | none | none | The pod is rejected since it does not enumerate a request. |
 
-The server is updated to be aware of **ResourceQuota** objects.
+The rationale for accounting for the requested amount of a resource versus the limit is the belief
+that a user should only be charged for what they are scheduled against in the cluster.  In addition,
+attempting to track usage against actual usage, where request < actual < limit, is considered highly
+volatile.
 
-The quota is only enforced if the kube-apiserver is started as follows:
+As a consequence of this decision, the user is able to spread its usage of a resource across multiple tiers
```

Comment:
Nit: Just to be consistent, do we want to call these "classes" instead of service? I don't think it's a big issue